qiskit/extensions/standard/rx.py

```diff
@@ -64,3 +83,48 @@ def rx(self, theta, q):  # pylint: disable=invalid-name
 
 
 QuantumCircuit.rx = rx
+
+
+class CrxGate(ControlledGate):
+    """controlled-rx gate."""
+
+    def __init__(self, theta):
+        """Create new crx gate."""
+        super().__init__('crx', 2, [theta], num_ctrl_qubits=1)
+        self.base_gate = RXGate
+        self.base_gate_name = 'rx'
```

Comment:
It appears like this can be removed here as well as for the drawer. I'll remove it from the class and keep it as a property with a deprecation warning.

Reply:
That was bothering me too during the renaming of the gates. That could be resolved if the name were a static property, but since currently the name attribute is set at runtime through `super.__init__(name)` I don't think that's possible...

Reply:
So there are a couple issues:

1- I think the `base_gate` should be the instance, not the class. The class is too generic; here we are controlling a specific instance of that class. For example I may want to control a circuit by doing `circuit.to_gate().control()`. Here `base_gate` currently just becomes `Gate` which does not seem very useful. Instead if it were the instance, I could immediate find the base and work with it.

2- I agree with @Cryoris that some of the Gate attributes that are currently instance properties should become class properties. For example the `U1Gate` class should know that it is a 1-qubit gate, whereas currently this information does not exist until an instance is created. 

Let's discuss these as a follow-up.

Reply:
This was implemented as a class as requested in #2862 for an optimization pass implemented in #3185. It could have been implemented as an instance but the class method requested seemed to pose less risk of bugs or performance issues. Perhaps those can be resolved with some care.

I agree with the class attributes where possible e.g. matrix representation.